CompilerSource/compiler/components/parse_and_link.cpp

```diff
@@ -245,26 +249,15 @@ int lang_CPP::compile_parseAndLink(const GameData &game, CompileState &state) {
       int ev_id;
         if (event.has_name()) {
           edbg << "  Event[" << event.type() << ", " << event.name() << "] ";
-          user << "FATAL: Unsupported: Event names. Translation required using "
-                  "events.res. Translation is present but there is no logic to "
-                  "use it.";
-
-          // TODO: Using the event type from events.res, translate the name
-          // string to a number and a valid identifier for use in this routine.
-          // It may be wise to store this information in a normalized table
-          // within GameData, similar to how fonts store glyph metrics.
```

Comment:
What this comment meant was, rather than doing lookups on the fly in the compiler, perform them in the GameData constructor, storing this translated information separately. This prevents error-prone translations and surfaces problems earlier.